src/openstack_cloud/__init__.py

```diff
@@ -17,6 +19,52 @@
 CLOUDS_YAML_PATH = Path(Path.home() / ".config/openstack/clouds.yaml")
 
 
+# Make sure we can import openstack, if not remove the old openstacksdk library and retry.
+# This is a workaround for https://bugs.launchpad.net/juju/+bug/2058335
+def _remove_old_openstacksdk_lib() -> None:
+    """Remove the old openstacksdk library if it exists."""
+    try:
+        unit_name = os.environ["JUJU_UNIT_NAME"].replace("/", "-")
+        venv_dir = Path(f"/var/lib/juju/agents/unit-{unit_name}/charm/venv/")
+        openstacksdk_dirs = list(venv_dir.glob("openstacksdk-*.dist-info"))
+        # use error log level as logging may not be fully initialized yet
+        logger.error("Found following openstack dirs: %s", openstacksdk_dirs)
+        if len(openstacksdk_dirs) > 1:
+            openstacksdk_dirs.sort()
```

Comment:
completely out of curiosity: why sort?

Reply:
out of curiosity again: could rmtree with ignore error=true be used?

Reply:
Ahhh I thought that the library would be installed after removing existing ones. Nice :)